src/publish/common/publish.ts

```diff
@@ -237,14 +238,14 @@ export async function handlePublish<
 }
 
 export async function renderForPublish(
-  render: (siteUrl?: string) => Promise<PublishFiles>,
+  render: (flags?: RenderFlags) => Promise<PublishFiles>,
   providerName: string,
   type: "document" | "site",
   title: string,
   siteUrl?: string,
 ) {
   // render
-  let publishFiles = await render(siteUrl);
+  let publishFiles = await render({ siteUrl });
```

Comment:
I'm a little worried about this one.  I'm checking it out...

Reply:
It seems good.  The GH pages reference seems to use the flags as well.  

https://github.com/quarto-dev/quarto-cli/blob/0da4a0f5acc07291d9cbccb6ce78d78f9ef6dbf5/src/publish/gh-pages/gh-pages.ts#L105